rust/automerge-c/src/item.rs

```diff
@@ -1241,26 +1241,26 @@ pub enum AMvalType {
     F64 = 1 << 8,
     /// A 64-bit signed integer value.
     Int = 1 << 9,
+    /// A mark
+    Mark = 1 << 10,
     /// A null value.
```

Comment:
@jkankiewicz It's good practice when adding to an enum to keep the old values the same. Although dependents shouldn't depend on the actual value, it makes life easier if they could.

Reply:
> It's good practice when adding to an enum to keep the old values the same.

@ConradIrwin, it's good practice when there's a legacy unversioned serialization format because there's no way to ensure that a C enum's tag will be deserialized from its previously- corresponding integer value otherwise. In this case, however, a C enum's tag is translated into its corresponding Rust enum's variant prior to serialization so its corresponding integer value is irrelevant.

Furthermore, I try to keep a C enum's tags sorted in alphabetical order because, unlike with Rust's `match` expression, the C compiler will not report an error when a `case` for one or more tags in an enum is missing from a `switch` statement on a value of that enum. It's easier for one to notice that a tag has been neglected when they're in alphabetical order.
While I could retain the subsequent tags' previous bit assignments despite the insertion of a new tag in alphabetical order, it's helpful when debugging for the adjacency relationships between bits in a bitmask to mirror the adjacency relationships between their corresponding tags.

> Although dependents shouldn't depend on the actual value, it makes life easier if they could.

I agree that they shouldn't and I would add that dependents should be strongly encouraged to prefer named constants over magic numbers.